experimental/skills/DialogToDialog/DialogRootBot/Dialogs/SkillDialog.cs

```diff
@@ -0,0 +1,153 @@
+            }
+
+            var skillId = dialogArgs.SkillId;
+            if (!_skillsConfig.Skills.TryGetValue(skillId, out var skillInfo))
+            {
+                throw new KeyNotFoundException($"Unable to find \"{skillId}\" in the skill configuration.");
+            }
+
+            // Store Skill ID for this dialog instance
+            await _activeSkillProperty.SetAsync(dc.Context, skillId, cancellationToken);
+
+            await dc.Context.SendTraceActivityAsync($"{GetType().Name}.BeginDialogAsync(). Using activity of type: {dialogArgs.ActivityType}", cancellationToken: cancellationToken);
+
+            Activity skillActivity;
+            switch (dialogArgs.ActivityType)
+            {
+                case ActivityTypes.Event:
+                    var eventActivity = Activity.CreateEventActivity();
+                    eventActivity.Name = dialogArgs.Name;
+                    eventActivity.ApplyConversationReference(dc.Context.Activity.GetConversationReference(), true);
+                    skillActivity = (Activity)eventActivity;
+                    break;
+
+                case ActivityTypes.Message:
+                    var messageActivity = Activity.CreateMessageActivity();
+                    messageActivity.Text = dialogArgs.Text;
+                    skillActivity = (Activity)messageActivity;
+                    break;
+
+                default:
+                    throw new ArgumentException($"Invalid activity type in {dialogArgs.ActivityType} in {nameof(SkillDialogArgs)}");
+            }
+
+            ApplyParentActivityProperties(dc, skillActivity, dialogArgs);
+            return await SendToSkill(dc, skillActivity, skillInfo, cancellationToken);
+        }
+
+        public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
+        {
+            await dc.Context.SendTraceActivityAsync($"{GetType().Name}.ContinueDialogAsync(). ActivityType: {dc.Context.Activity.Type}", cancellationToken: cancellationToken);
+
+            var skillId = await _activeSkillProperty.GetAsync(dc.Context, () => null, cancellationToken);
+
+            if (dc.Context.Activity.Type == ActivityTypes.Message && dc.Context.Activity.Text.Equals("abort", StringComparison.CurrentCultureIgnoreCase))
+            {
+                // Send a message to the skill to let it do some cleanup
+                var eocActivity = Activity.CreateEndOfConversationActivity();
+                eocActivity.ApplyConversationReference(dc.Context.Activity.GetConversationReference(), true);
+                await SendToSkill(dc, (Activity)eocActivity, _skillsConfig.Skills[skillId], cancellationToken);
+
+                // End this dialog and return (we don't care if the skill responds or not)
+                await dc.Context.SendTraceActivityAsync($"{GetType().Name}.ContinueDialogAsync(). Cancelled", cancellationToken: cancellationToken);
```

Comment:
```suggestion
                await dc.Context.SendTraceActivityAsync($"{GetType().Name}.ContinueDialogAsync(). Canceled", cancellationToken: cancellationToken);
```

Reply:
OK, changed it, however, our DialogTurnStatus constant uses the British from "Cancelled" 

---
In reply to: [356230933](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356230933) [](ancestors = 356230933)